### ISSUE TITLE
Macro expansion in external dependencies fails when combined with `OTHER_SWIFT_FLAGS`

### ISSUE THREAD

**Issue by thebarndog** [tuist/tuist#5944]
### What happened?

I'm building an application that uses both RealmSwift and ComposableArchitecture. When dealing with Swift concurrency, Realm recommends enabling the `-Xfrontend-enable-actor-data-race-checks` Swift flag. However, when that flag is set in conjunction to using a package that exports Swift macros, building inside the generated Xcode project fails.

I provided a sample project that illustrates this issue but essentially I was able to narrow down the problem line to

```swift
let settings: SettingsDictionary = [:].otherSwiftFlags("-enable-actor-data-race-checks")
```

Without that other swift flag, my project compiles. Maybe I'm including the flag incorrectly? Project generation works but building the generated project breaks. I'm not sure what the underlying issue is but I imagine it'll be an important one with the advent of Swift 6 and more strict concurrency checks. 

[Sample Project](https://github.com/tuist/tuist/files/14317704/Sample.zip)

### How do we reproduce it?

1. Run `tuist install` on the provided sample project to install ComposableArchitecture
2. Run `tuist generate`
3. Attempt to build
4. Observe error  `External macro implementation type 'ComposableArchitectureMacros.ReducerMacro' could not be found for macro 'Reducer()'`

### Error log

```
External macro implementation type 'ComposableArchitectureMacros.ReducerMacro' could not be found for macro 'Reducer()'
```

### macOS version

14.1.2

### Tuist version

4.2.5

### Xcode version

15.2

**Comment by kdjun97** [comment]
I know it's the kind of error that doesn't recognize @reducer macro in tuist.
To resolve this error, u can do the following.

Don't use @reducer macro, but adopt Reducer protocol.

ex)

@Reducer
struct Feature { 
  ...
}

-> (do this example)

struct Feature: Reducer {
  ...

  var body: some Reducer<State, Action> {
    Redcue { state, action in 
       ...
    }
  }
}

**Comment by waltflanagan** [comment]
👋  this isnt super obvious, but if you update your settings to `[:].otherSwiftFlags("$(inherited)", "-enable-actor-data-race-checks")` things should be properly merged, instead of overwritten.  LMK if this solves your issues.  I can raise this to the team and see if there's anything we could do to make this more obvious since this is likely to be more of an issues as we approach Swift 6 and more macro adoption.

**Comment by fortmarek** [comment]
I am reopening this one. I think we should force the inclusion of `$(inherited)` if it would lead to the project not building cc @pepicrft 

It should not be a lot of code to achieve this imho.

**Comment by pepicrft** [comment]
@fortmarek sorry for the question, how do you see the inclusion being force?

**Comment by fortmarek** [comment]
> sorry for the question, how do you see the inclusion being force?

We can definitely add this into a macro mapper that updates the user-created settings and makes sure that the `$(inherited)` flag is there to ensure macros can be built.

**Comment by timoninas** [comment]
It doesn't solve the issue
I've tried to launch project with these options, but it didn't meet success

The error is the same `External macro implementation type 'MyMacros.BuilderMacro' could not be found for macro 'Builder()'`

**Comment by BakrIOS91** [comment]
Are you using non-standard configuration names in Xcode? Something other than Release or Debug? Then this may explain things https://github.com/pointfreeco/swift-composable-architecture/discussions/3292#discussioncomment-10592013

**Comment by fortmarek** [comment]
@timoninas the fix was released in [Tuist 4.6.0](https://github.com/tuist/tuist/releases/tag/4.6.0), so it's expected things don't work for you on Tuist 3.8.0. We recommend updating to the latest version.

If you are still running into the issue on the latest version, please open a new issue with a reproducible sample.
